app/workers/builds.php

```diff
@@ -251,17 +251,24 @@ protected function buildDeployment(GitHub $github, Document $project, Document $
 
                 $tmpPath = '/tmp/builds/' . \escapeshellcmd($buildId);
                 $tmpPathFile = $tmpPath . '/code.tar.gz';
+                $localDevice = new Local();
+
+                if (substr($tmpDirectory, -1) !== '/') {
+                    $tmpDirectory .= '/';
+                }
+
+                $directorySize = $localDevice->getDirectorySize($tmpDirectory);
+                $functionsSizeLimit = (int) App::getEnv('_APP_FUNCTIONS_SIZE_LIMIT', '30000000');
+                if ($directorySize > $functionsSizeLimit) {
+                    throw new Exception('Repository directory size should be less than ' . number_format($functionsSizeLimit / 1048576, 2) . ' MBs.');
+                }
```

Comment:
I'm a little concerned about the inconsistency between VCS and manual deployment. In manual deployment, the size limit is on the compressed archive:

https://github.com/appwrite/appwrite/blob/ddb73eb4f60aa807fd0bfe8622da32a69738077a/app/controllers/api/functions.php#L1082-L1120

but here, we're looking at the uncompressed directory size.

Reply:
With manual, compression is done by client. With VCS, compression is done by server.
I am worried we would be spending resources, allowing DOS attack if we moved it lower.

I understand inconsistency concerns, but I think only very very few devs should be hitting the limit on Cloud. And if hit on self-hosted, its so easy to increase that it doesn't matter too much if it's before or after compression size

Reply:
> I think only very very few devs should be hitting the limit on Cloud. And if hit on self-hosted

@Meldiron, I don't think it's comparison between cloud vs self-hosted. It's manual deployment vs VCS. Both can be done on cloud and self-hosted.